jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import mockit.Expectations;
+import mockit.Mocked;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jkube.kit.common.Dependency;
+import org.eclipse.jkube.kit.common.JavaProject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class SpringBootUtilTest {
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void testGetSpringBootApplicationProperties() throws IOException {
+
+        //Given
+        File applicationProp =  new File(getClass().getResource("/util/spring-boot-application.properties").getPath());
+        String springActiveProfile = null;
+        File targetFolder = temporaryFolder.newFolder("target");
+        File classesInTarget = new File(targetFolder, "classes");
+        boolean isTargetClassesCreated = classesInTarget.mkdirs();
+        File applicationPropertiesInsideTarget = new File(classesInTarget, "application.properties");
+        FileUtils.copyFile(applicationProp, applicationPropertiesInsideTarget);
+        URLClassLoader urlClassLoader = ClassUtil.createClassLoader(Arrays.asList(classesInTarget.getAbsolutePath(), applicationProp.getAbsolutePath()), classesInTarget.getAbsolutePath());
+
+        //When
+        Properties result =  SpringBootUtil.getSpringBootApplicationProperties(springActiveProfile ,urlClassLoader);
+
+        //Then
+        assertTrue(isTargetClassesCreated);
+        assertEquals("demoservice" ,result.getProperty("spring.application.name"));
+        assertEquals("9090" ,result.getProperty("server.port"));
+    }
+
+    @Test
+    public void testGetSpringBootDevToolsVersion(@Mocked JavaProject maven_project) {
+
+        //Given
+        Dependency p = new Dependency();
+        p.setGroupId("org.springframework.boot");
+        p.setVersion("1.6.3");
+        new Expectations() {{
+            maven_project.getDependencies();
+            result=p;
```

Comment:
I don't understand how this can be working, getDependencies() method returns a List, your mock returns an instance of Dependency.